ui/hooks/src/use-login.ts

```diff
@@ -267,8 +270,15 @@ export function useCheckSignup(ethAddress) {
   return useQuery(
     CHECK_SIGNUP_KEY,
     async () => {
-      const resp = await lastValueFrom(sdk.api.auth.checkIfSignedUp(ethAddress));
-      return resp.data;
+      try {
+        const resp = await lastValueFrom(sdk.api.auth.checkIfSignedUp(ethAddress));
+        if (resp.data && resp.data.hasOwnProperty('getProfile')) {
```

Comment:
this can be shorthanded like `return resp.data && resp.data.hasOwnProperty('getProfile')`